ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -181,7 +232,11 @@ def forward(
         encoded_self = self.observation_encoder(inputs)
         if actions is not None:
             encoded_self = torch.cat([encoded_self, actions], dim=1)
-        encoding = self.linear_encoder(encoded_self)
+        if isinstance(self.linear_encoder, ConditionalEncoder):
+            goal = self.observation_encoder.get_goal_encoding(inputs)
+            encoding = self.linear_encoder(encoded_self, goal)
```

Comment:
If we're going to do both Conditional and Linear I think we should rename it to just `encoder` or `body_encoder`, but not really that necessary